types/elasticsearch/index.d.ts

```diff
@@ -74,8 +74,11 @@ export class Client {
     renderSearchTemplate(params: RenderSearchTemplateParams): Promise<any>;
     scroll<T>(params: ScrollParams, callback: (error: any, response: SearchResponse<T>) => void): void;
     scroll<T>(params: ScrollParams): Promise<SearchResponse<T>>;
-    search<T>(params: SearchParams, callback: (error: any, response: SearchResponse<T>) => void): void;
-    search<T>(params: SearchParams): Promise<SearchResponse<T>>;
+    search<T, TTotalHits = number>(
+        params: SearchParams,
+        callback: (error: any, response: SearchResponse<T, TTotalHits>) => void,
+    ): void;
+    search<T, TTotalHits = number>(params: SearchParams): Promise<SearchResponse<T, TTotalHits>>;
```

Comment:
In the Elasticsearch client API 7.0 they changed the type of the `SearchResponse.hits.total` value from `number` to an object (that this PR names `TotalHits`), but they provided a new option, `restTotalHitsAsInt`, to preserve the old type in the results. In API 8.0, the `restTotalHitsAsInt` was removed and the results always use the new type. I've tried here to preserve the existing type behavior of the `search` method while allowing the type of the `hits.total` value to be overridden. If there is a better way to do this I am open to suggestions.